SeaPublicWebsite/ExternalServices/BreApi.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SeaPublicWebsite.DataModels;
+using SeaPublicWebsite.ExternalServices.Models;
+using SeaPublicWebsite.Helpers;
+using SeaPublicWebsite.Services;
+
+namespace SeaPublicWebsite.ExternalServices
+{
+    public static class BreApi
+    {
+        public static async Task<List<Recommendation>> GetRecommendationsForUserRequest(BreRequest request)
+        {
+            try
+            {
+                using (HttpClient httpClient = new HttpClient())
+                {
+                    httpClient.BaseAddress = new Uri(Global.BreBaseAddress);
+
+                    string username = Global.BreUsername;
+                    string password = Global.BrePassword;
+                    Guid nonce =  Guid.NewGuid();
+                    string created = DateTime.Now.ToUniversalTime().ToString
+                        (DateTimeFormatInfo.InvariantInfo.SortableDateTimePattern) + "Z";
+                    string token = GenerateToken(password + nonce + username + created);
+                    string wsseHeader =
+                        $"WSSE UsernameToken Token=\"{token}\", Nonce=\"{nonce}\", Username=\"{username}\", Created=\"{created}\"";
+                    httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", wsseHeader);
+
+                    string path = "/bemapi/energy_use";
+                    string requestString = JsonConvert.SerializeObject(request);
+                    StringContent stringContent = new(requestString);
+                    HttpResponseMessage response = await Task.FromResult(httpClient.PostAsync(path, stringContent)).Result;
```

Comment:
This should just be `HttpResponseMessage response = await httpClient.PostAsync(path, stringContent);`
Same below